packages/cli/src/rpc/navie/services/threadIndexService.ts

```diff
@@ -7,6 +7,7 @@ import { mkdirSync } from 'node:fs';
 
 const INITIALIZE_SQL = `
 PRAGMA foreign_keys = ON;
+PRAGMA journal_mode = WAL;
```

Comment:
Are you sure this is the problem with locks? I think it might have been (perhaps in addition to this) that you're preparing statements but then never disposing of them. I've seen this leave stale locks around after the program exits and requiring manual removal of lock files in order to be able to open the database.

In general, I don't think it makes sense to hold prepared statements long-term like this. If it's not in a hot loop, just execute it without preparing. If it is, prepare it before the loop and dispose after.

(Also in some cases it actually can be more efficient, since the statistics might have changed since the program started and a different query plan is better now.)